kairon/actions/definitions/store_page.py

```
@@ -0,0 +1,78 @@
+
+    def __init__(self, bot: Text, name: Text):
+        self.bot = bot
+        self.name = name
+
+    def retrieve_config(self):
+        try:
+            return StorePageAction.objects(
+                bot=self.bot, name=self.name, status=True
+            ).get().to_mongo().to_dict()
+        except DoesNotExist:
+            raise ActionFailure("No StorePageAction found for given action and bot")
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)